### ISSUE TITLE
val_loss is calculated differently when using multiple output model

### ISSUE THREAD

**Issue by vladostan** [keras-team/keras#13838]
I'm interested in way how keras calculates val_loss when model have multiple outputs. Let's say that I have 6 softmax outputs in my model. BCE is the loss function used for each of the outputs. I've assigned equal weights for each output.

        loss = {
            "softmax_0": 'binary_crossentropy',
            "softmax_1": 'binary_crossentropy',
            "softmax_2": 'binary_crossentropy',
            "softmax_3": 'binary_crossentropy',
            "softmax_4": 'binary_crossentropy',
            "softmax_5": 'binary_crossentropy'
        }

        loss_weights = {
                "softmax_0": 1.,
                "softmax_1": 1.,
                "softmax_2": 1.,
                "softmax_3": 1.,
                "softmax_4": 1.,
                "softmax_5": 1.
        }

The model is compiled as follows:

model.compile(optimizer=optimizer, loss=loss, loss_weights=loss_weights, metrics=["accuracy"])

When I train my model, the training behavior is as expected, i.e. total loss is equal to sum of all six losses. But during validation, the val_loss is calculated wrongly, I suppose.

Example:

loss: 4.2090 - softmax_0_loss: 0.5218 - softmax_1_loss: 0.7349 - softmax_2_loss: 0.7294 - softmax_3_loss: 0.8700 - softmax_4_loss: 0.8834 - softmax_5_loss: 0.4695 - softmax_0_accuracy: 0.8558 - softmax_1_accuracy: 0.7927 - softmax_2_accuracy: 0.7927 - softmax_3_accuracy: 0.7392 - softmax_4_accuracy: 0.7335 - softmax_combined_accuracy: 0.8720 - val_loss: 1.0564 - val_softmax_0_loss: 0.2044 - val_softmax_1_loss: 0.2865 - val_softmax_2_loss: 0.2701 - val_softmax_3_loss: 0.4560 - val_softmax_4_loss: 0.3901 - val_softmax_5_loss: 0.2045 - val_softmax_0_accuracy: 0.9037 - val_softmax_1_accuracy: 0.8676 - val_softmax_2_accuracy: 0.8785 - val_softmax_3_accuracy: 0.7414 - val_softmax_4_accuracy: 0.8244 - val_softmax_combined_accuracy: 0.8969

Here the loss is 4.2090, and the sum of six train losses is 0,7349+0,7294+0,87+0,8834+0,4695+0,5218 = 4.209. val_loss should be 0,2044+0,2865+0,2701+0,4560+0,3901+0,2045 = 1.8116, but it is 1.0564 for some reason.

Could someone please explain such behavior?

**System information**  
- TensorFlow version:  1.14.0
- Keras version:  2.3.1
- Python version:  3.6.7
- CUDA/cuDNN version:  10.0/7.4.2
- GPU model and memory:  Tesla V100, 16GB

**Comment by PalashJauhari** [comment]
I also observed similar behavior while training keras multiple output , multiclass model. For train dataset individual losses are summing to total loss, but for val_loss it doesnt sums up. Any leads on this?

**Comment by sushreebarsa** [comment]
@vladostan Moving this issue to closed status as there has been no recent activity, in case you still face the error please create a new issue.Thanks!
